### PR TITLE
fix: close messageInfo when message not in redux store

### PR DESCRIPTION
**Pull request by Bilb**

### CODE REVIEW

File: ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx [hunk 1]
```diff
@@ -78,36 +95,133 @@ const StyledMessageDetail = styled.div`
   padding: var(--margins-sm) var(--margins-2xl) var(--margins-lg);
 `;
 
+type MessagePropsDetails = {
```
Comment (yougotwill) [inline ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx:98]: rename to MessageInfoProps for consistency?

File: ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx [hunk 2]
```diff
@@ -78,36 +95,133 @@ const StyledMessageDetail = styled.div`
...
+  errors: Array<Error>;
+  attachments: Array<PropsForAttachment>;
+};
+
+async function getPropsForMessageDetail(
```
Comment (yougotwill) [inline ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx:103]: getPropsForMessageInfo?

File: ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx [hunk 3]
```diff
@@ -78,36 +95,133 @@ const StyledMessageDetail = styled.div`
...
+      errors,
+      attachments: attachmentsWithMediaDetails,
+    };
+
+    return toRet;
+  }
+  return null;
+}
+
+function useMessageDetailProps(messageId: string | undefined): MessagePropsDetails | null {
```
Comment (yougotwill) [inline ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx:162]: MessageInfo? 🙃

Comment (yougotwill) [reply ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx:162]: Same on line 164 lol

File: ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx [hunk 4]
```diff
@@ -78,36 +95,133 @@ const StyledMessageDetail = styled.div`
...
+  const [details, setDetails] = useState<MessagePropsDetails | null>(null);
+
+  const fromState = useMessageAttachments(messageId);
+
+  // this is not ideal, but also doesn't seem to create any performance issue at the moment.
+  // TODO: ideally, we'd want to save the attachment duration anytime we save one to the disk (incoming/outgoing), and just retrieve it from the redux state here.
```
Comment (yougotwill) [inline ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx:168]: save to redux only or save to the db?

File: ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx [hunk 5]
```diff
@@ -78,36 +95,133 @@ const StyledMessageDetail = styled.div`
...
+  useEffect(() => {
```
Comment (yougotwill) [inline ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx:169]: Correct me if I am wrong but is this not a case for `useMount`?

File: ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx [hunk 6]
```diff
@@ -78,36 +95,133 @@ const StyledMessageDetail = styled.div`
...
+  const messageId = useSelector(getMessageInfoId);
+  const messageDetailProps = useMessageDetailProps(messageId);
+  const isDeletable = useMessageIsDeletable(messageId);
+  const direction = useMessageDirection(messageId);
+  const timestamp = useMessageTimestamp(messageId);
+  const serverTimestamp = useMessageServerTimestamp(messageId);
+  const sender = useMessageSender(messageId);
 
-  const dispatch = useDispatch();
+  // we close the right panel when switching conversation so the convoId of the convoId of that message
```
Comment (yougotwill) [inline ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx:200]: I'm guessing this should be `convoId of the convo`?

File: ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx [hunk 7]
```diff
@@ -78,36 +95,133 @@ const StyledMessageDetail = styled.div`
...
+  // is always the currently selected conversation
+  const convoId = useSelectedConversationKey();
 
-  useKey('Escape', () => {
+  const closePanel = useCallback(() => {
     dispatch(closeRightPanel());
     dispatch(resetRightOverlayMode());
-    dispatch(closeMessageDetailsView());
-  });
+    dispatch(closeMessageInfoView());
```
Comment (yougotwill) [inline ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx:207]: Shouldn't resetting the overlay mode clear the message info view already?

File: ts/components/conversation/right-panel/overlay/message-info/components/MessageInfo.tsx [hunk 8]
```diff
@@ -51,16 +57,18 @@ const showDebugLog = () => {
   ipcRenderer.send('show-debug-log');
 };
 
-export const MessageInfo = () => {
-  const messageDetailProps = useSelector(getMessageDetailsViewProps);
+export const MessageInfo = ({ messageId, errors }: { messageId: string; errors: Array<any> }) => {
```
Comment (yougotwill) [inline ts/components/conversation/right-panel/overlay/message-info/components/MessageInfo.tsx:60]: typing for the `any`?

File: ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx (hunk 7 above)
Comment (yougotwill) [reply ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx:207]: I looked at the function. Shouldn't we just clear that idea in the `resetRightOverlayMode` function? That way we don't need two functions.

**Review by yougotwill — CHANGES_REQUESTED** [review]

File: ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx (hunk 4 above)
Comment (Bilb) [reply ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx:168]: to the DB

File: ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx (hunk 5 above)
Comment (Bilb) [reply ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx:169]: I don't think so.
I think useMount wouldn't reload the data if we keep the right panel open but switch between two messages right

File: ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx (hunk 6 above)
Comment (Bilb) [reply ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx:200]: fixed to 

  // we close the right panel when switching conversation so the convoId of that message is always the selectedConversationKey

File: ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx (hunk 4 above)
Comment (Bilb) [reply ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx:168]: there is no need to keep refreshing those details. Given an attachment, they are the same for the whole life of that attachment

File: ts/components/conversation/right-panel/overlay/message-info/components/MessageInfo.tsx (hunk 8 above)
Comment (Bilb) [reply ts/components/conversation/right-panel/overlay/message-info/components/MessageInfo.tsx:60]: fixed

File: ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx (hunk 7 above)
Comment (Bilb) [reply ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx:207]: updated

File: ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx (hunk 4 above)
Comment (yougotwill) [reply ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx:164]: Got it. Resolved

File: ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx (hunk 5 above)
Comment (yougotwill) [reply ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx:165]: True. Resolved

**Review by yougotwill — APPROVED** [review]
LGTM 🛳️

**Pull request merged**